File: Crio/crio_programming_interview_problems-master/MoveMiddleToHead/MoveMiddleToHead.java
/*
public class ListNode {
   public int val; 
   public ListNode next;
   public ListNode(int x) { val = x; next = null; }
}
*/
import java.util.*;
import crio.ds.List.ListNode;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

class MoveMiddleToHead{
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt();
		ArrayList<Integer> input=new ArrayList<Integer>();
		for(int i=0;i<n;i++)
			input.add(sc.nextInt());
		ListNode head=ListNode.createList(input);
		Solution so=new Solution();
		ListNode start=so.moveMiddleToHead(head);
		List<Integer> ans=ListNode.extractList(start);
		for(int i=0;i<ans.size();i++)
		{
			System.out.print(ans.get(i)+" ");
		} 

	}
}

File: Crio/LinkedList/6_MoveMiddleToHead/MoveMiddleToHead.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"
#include "../crio/cpp/io/FastIO.hpp"

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

int main() {

	FastIO();
	int n;
	cin >> n;
	vector<int> list(n);
	for ( int i = 0; i < n; i++) cin >> list[i];

	ListNode * head = ListNode().createList(list);
	head = moveMiddleToHead(head);
	vector<int> result = ListNode().extractList(head);
	for ( int it : result) {
		cout << it << " ";
	}
}
